Lib/test/test_c_locale_coercion.py

```diff
@@ -143,27 +143,29 @@ def test_library_c_locale_warning(self):
     "or PYTHONCOERCECLOCALE=0 to disable this locale coercion behavior)."
 )
 
-class _LocaleCoercionTargetsTestCase(_ChildProcessEncodingTestCase):
-    # Base class for test cases that rely on coercion targets being defined
 
-    available_targets = []
-    targets_required = True
+AVAILABLE_TARGETS = None
 
-    @classmethod
-    def setUpClass(cls):
-        first_target_locale = None
-        available_targets = cls.available_targets
-        # Find the target locales available in the current system
-        for target_locale in _C_UTF8_LOCALES:
-            if _set_locale_in_subprocess(target_locale):
-                available_targets.append(target_locale)
-                if first_target_locale is None:
-                    first_target_locale = target_locale
-        if cls.targets_required and not available_targets:
-            raise unittest.SkipTest("No C-with-UTF-8 locale available")
-        # Expect coercion to use the first available locale
-        warning_msg = CLI_COERCION_WARNING_FMT.format(first_target_locale)
-        cls.EXPECTED_COERCION_WARNING = warning_msg
+def setUpModule():
+    global AVAILABLE_TARGETS
+
+    if AVAILABLE_TARGETS is not None:
+        # initialization already done
+        return
+    AVAILABLE_TARGETS = []
+
+    # Find the target locales available in the current system
+    for target_locale in _C_UTF8_LOCALES:
+        if _set_locale_in_subprocess(target_locale):
+            AVAILABLE_TARGETS.append(target_locale)
+    if not AVAILABLE_TARGETS:
+        raise unittest.SkipTest("No C-with-UTF-8 locale available")
+
+
+
+class _LocaleCoercionTargetsTestCase(_ChildProcessEncodingTestCase):
+    # Base class for test cases that rely on coercion targets being defined
+    pass
```

Comment:
The `cls.targets_required and not AVAILABLE_TARGETS` check should still be handled in `setupClass` so that the other tests still run as expected, even when locale coercion isn't possible.